src/realm.h

```diff
@@ -1872,4 +1872,40 @@ RLM_API realm_notification_token_t* realm_results_add_notification_callback(real
  */
 RLM_API realm_results_t* realm_results_from_thread_safe_reference(const realm_t*, realm_thread_safe_reference_t*);
 
+/* Logging */
+typedef struct realm_logger realm_logger_t;
+
+typedef enum realm_log_level {
+    RLM_LOG_LEVEL_ALL,
```

Comment:
Related to other comments, make sure to give these explicit values and a comment to which other type they must match. And at this point why even have different types :-)

Reply:
They need to be different types because the C header needs to have an air-gap with everything else. It cannot include any of our other files (in part because they are written in C++!) and no files outside of the c_api can include this (because the C API is supposed to be a wrapper around the C++ API).